services/QuillLMS/lib/tasks/vitally.rake

```diff
@@ -19,7 +19,12 @@ namespace :vitally do
     # has a batch limit of 100.
     CSV.parse(pipe_data, headers: true).each_slice(33) do |batch|
       batch_payload = batch.map do |row|
-        user = User.includes(:schools_users).find(row['externalId'])
+        # TIL `includes` makes a separate query to the db, and `joins` doesn't store the joined value in ActiveRecord, so if you want a single query to pull multiple models you need to use both?
+        user = User.joins(:schools_users).includes(:schools_users).find_by(id: row['externalId'])
+
+        # Some of the data provided by Vitally is bad, so we need to account for that
+        next unless user
+
```

Comment:
Very minor, but if we're adding a joins, do is safe navigation necessary here:
 `quill_school_id = user.schools_users&.school_id`

Reply:
Good catch.  We actually want cases where there are no school assignments, so I think I actually need to make the `joins` into a `left_outer_joins`.